pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -91,6 +92,7 @@ func (configgen *ConfigGeneratorImpl) buildGatewayListeners(
 				if gateway.IsTLSServer(server) && gateway.IsHTTPServer(server) {
 					// This is a HTTPS server, where we are doing TLS termination. Build a http connection manager with TLS context
 					routeName := mergedGateway.RouteNamesByServer[server]
+					node.GatewayRouteNames = append(node.GatewayRouteNames, routeName)
```

Comment:
it is possible to patch a route

Reply:
true. here, https://github.com/istio/istio/pull/20124/files#diff-f507fb55b7edf4a03d3c7e9aebf64a1eR197, patched routes will also be there in `rawRoutes` and then below that we are collecting clusters from those as well.